FrontEndReact/src/View/Login/SetNewPassword.js

```diff
@@ -3,87 +3,264 @@ import 'bootstrap/dist/css/bootstrap.css';
+            if (atLeastOneUppercaseLetter(password)) points += 1;
+            if (atLeastOneLowercaseLetter(password)) points += 1;
+            if (atLeastOneNumber(password)) points += 1;
+            if (atLeastOneSpecialChar(password)) points += 1;
+        
+            if (points >= 5) return 'STRONG';
+            if (points >= 3) return 'MEDIUM';
+
+            return 'WEAK';
         }
+        
+
         this.setPassword = () => {
-            var pass1 = document.getElementById("password").value;
-            var pass2 = document.getElementById("password2").value;
-    
-            if (pass1 === pass2) {
+            var pass1 = this.state.password;
+            var pass2 = this.state.confirmationPassword;
 
-                var valid = validPasword(pass1);
+            var passwordSecurity = this.testPasswordStrength(pass1);
 
-                if(valid === true)
-                {
-                    let body = JSON.stringify({
-                        'password': pass1
-                        })
-                    
-                    genericResourcePUT("/password", this, body)
-                    
-                    this.setState(() =>( {
-                        passSet: true
-                    }))
-                }
-                else 
-                {
+            if (pass1 === pass2) {
+                if (passwordSecurity !== "STRONG") {
                     this.setState(() => ({
-                        errorMessage: "Password must have " + valid
-                    }))
-                }              
-            }
-            else {
+                    errorMessage: "Please verify your password strength"
+                    }));
+                }
+                else {
+                    if (validPasword(pass1)) {
+                        let body = JSON.stringify({
+                            'password': pass1
+                        })
+    
+                        genericResourcePUT("/password", this, body)
+    
+                        this.setState(() =>({ isPasswordSet: true }));
+    
+                    } 
+                }  
+            } else {
                 this.setState(() => ({
-                    errorMessage: "Passwords do not match"
-                }))
-            }
+                    errorMessage: "Passwords do not match",
 
+                    errors: {
+                        password: "Passwords do not match",
+                        confirmationPassword: "Passwords do not match"
+                    }
+                }));
+            }
         }
     }
+
     render() {
         const {
             errorMessage,
-            passSet
+            isPasswordSet,
+            password,
+            confirmationPassword,
+            errors
         } = this.state;
+       
+        const passwordStrength = this.testPasswordStrength(password)
+        const colors = this.generateColors(passwordStrength);
+        const Icon = this.getIcon(passwordStrength);
```

Comment:
I suggest renaming `Icon` -> `icon`. Uppercase names are usually reserved for ADTs, classes, etc.

Reply:
In this instance Icon is storing a component corresponding to the password strength. Therefore, I am going to leave the capitalized Icon variable name!